diff --git a/webrtc-sys/Cargo.toml b/webrtc-sys/Cargo.toml
index d1bbd8bd..222300af 100644
--- a/webrtc-sys/Cargo.toml
+++ b/webrtc-sys/Cargo.toml
@@ -1,21 +1,21 @@
 [package]
 name = "webrtc-sys"
 version = "0.3.28"
-edition.workspace = true
+edition = "2021"
 homepage = "https://livekit.io"
-license.workspace = true
+license = "Apache-2.0"
 description = "Unsafe bindings to libwebrtc"
-repository.workspace = true
+repository = "https://github.com/livekit/rust-sdks"
 
 [features]
 default = []
 
 [dependencies]
 cxx = "1.0"
-log = { workspace = true }
+log = "0.4"
 
 [build-dependencies]
-webrtc-sys-build = { workspace = true }
+webrtc-sys-build = "0.3.15"
 cxx-build = "1.0"
 glob = "0.3"
 cc = "1.0"
@@ -25,4 +25,4 @@ pkg-config = "0.3.22"
 pkg-config = "0.3.22"
 
 [dev-dependencies]
-env_logger = { workspace = true }
+env_logger = "0.11"
diff --git a/webrtc-sys/build.rs b/webrtc-sys/build.rs
index 072794ec..d05b4082 100644
--- a/webrtc-sys/build.rs
+++ b/webrtc-sys/build.rs
@@ -23,7 +23,8 @@ fn main() {
 
     let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
     let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
-    let is_desktop = target_os == "linux" || target_os == "windows" || target_os == "macos";
+    // let is_desktop = target_os == "linux" || target_os == "windows" || target_os == "macos";
+    let is_desktop = false;
 
     println!("cargo:rerun-if-env-changed=LK_DEBUG_WEBRTC");
     println!("cargo:rerun-if-env-changed=LK_CUSTOM_WEBRTC");
@@ -168,11 +169,13 @@ fn main() {
             println!("cargo:rustc-link-lib=dylib=pthread");
             println!("cargo:rustc-link-lib=dylib=m");
 
-            // In order to avoid any ABI mismatches we use the sysroot's headers.
-            add_gio_headers(&mut builder);
+            if is_desktop {
+                // In order to avoid any ABI mismatches we use the sysroot's headers.
+                add_gio_headers(&mut builder);
 
-            for lib_name in ["glib-2.0", "gobject-2.0", "gio-2.0"] {
-                pkg_config::probe_library(lib_name).unwrap();
+                for lib_name in ["glib-2.0", "gobject-2.0", "gio-2.0"] {
+                    pkg_config::probe_library(lib_name).unwrap();
+                }
             }
 
             add_lazy_load_so(
diff --git a/webrtc-sys/libwebrtc/build_linux.sh b/webrtc-sys/libwebrtc/build_linux.sh
index 926e21be..05586d12 100755
--- a/webrtc-sys/libwebrtc/build_linux.sh
+++ b/webrtc-sys/libwebrtc/build_linux.sh
@@ -72,6 +72,9 @@ git apply "$COMMAND_DIR/patches/ssl_verify_callback_with_native_handle.patch" -v
 git apply "$COMMAND_DIR/patches/add_deps.patch" -v --ignore-space-change --ignore-whitespace --whitespace=nowarn
 git apply "$COMMAND_DIR/patches/fix_desktop_capture_compile.patch" -v --ignore-space-change --ignore-whitespace --whitespace=nowarn
 
+git apply "$COMMAND_DIR/patches/hooks-opus.patch" -v --ignore-space-change --ignore-whitespace --whitespace=nowarn
+git apply "$COMMAND_DIR/patches/hooks-vp9.patch" -v --ignore-space-change --ignore-whitespace --whitespace=nowarn
+
 # Disable CREL (compact relocations). Chromium's build enables experimental
 # CREL via -Wa,--crel which causes segfaults on aarch64-linux (and is known
 # broken on arm32 and s390x too).
@@ -116,12 +119,12 @@ args="is_debug=$debug  \
   enable_stripping=true \
   ffmpeg_branding=\"Chrome\" \
   rtc_use_h264=true \
-  rtc_use_h265=true \
-  rtc_use_pipewire=true \
+  rtc_use_h265=false \
+  rtc_use_pipewire=false \
   symbol_level=0 \
   enable_iterator_debugging=false \
   use_rtti=true \
-  rtc_use_x11=true"
+  rtc_use_x11=false"
 
 # generate ninja files
 gn gen "$OUTPUT_DIR" --root="src" --args="${args}"
